Samples/TrendingMovies/TrendingMovies/AppDelegate.swift

```diff
@@ -0,0 +1,166 @@
+            sharedCache.cleanExpiredDiskCache()
+            URLCache.shared.removeAllCachedResponses()
+        }
+
+        return true
+    }
+
+    func application(_: UIApplication, didFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        print("[TrendingMovies] didFinishLaunchingWithOptions")
+        Tracer.setUp(finishedLaunching: true)
+
+        window = UIWindow(frame: UIScreen.main.bounds)
+        let tabBarController = createTabBarController(items: [
+            TabBarItemSpec(
+                createViewController: createTrendingViewController,
+                title: Titles.trending,
+                icon: UIImage(named: "Trending")
+            ),
+            TabBarItemSpec(
+                createViewController: createNowPlayingViewController,
+                title: Titles.nowPlaying,
+                icon: UIImage(named: "NowPlaying")
+            ),
+            TabBarItemSpec(
+                createViewController: createUpcomingViewController,
+                title: Titles.upcoming,
+                icon: UIImage(named: "Upcoming")
+            ),
+        ])
+        window?.rootViewController = tabBarController
+        window?.makeKeyAndVisible()
+
+        // TODO: show debug menu launcher?
+
+        if ProcessInfo().arguments.contains("--io.sentry.ui-test.benchmarking") {
+            let vc = UIViewController(nibName: nil, bundle: nil)
+            let views = [benchmarkValueTextField, benchmarkRetrieveValueButton]
+            let stack = UIStackView(arrangedSubviews: views)
+            stack.translatesAutoresizingMaskIntoConstraints = false
+            stack.axis = .vertical
+            vc.view.addSubview(stack)
+            for view in views {
+                view.translatesAutoresizingMaskIntoConstraints = false
+                NSLayoutConstraint.activate([
+                    view.widthAnchor.constraint(equalToConstant: 100),
+                    view.heightAnchor.constraint(equalToConstant: 50),
+                ])
+            }
+            NSLayoutConstraint.activate([
+                stack.leadingAnchor.constraint(equalTo: vc.view.leadingAnchor),
+                stack.topAnchor.constraint(equalTo: vc.view.topAnchor)
+            ])
+            benchmarkValueTextField.accessibilityLabel = "io.sentry.accessibility-identifier.benchmarking-value-marshaling-text-field"
+            benchmarkRetrieveValueButton.accessibilityLabel = "io.sentry.accessibility-identifier.benchmarking-value-marshaling-button"
+            benchmarkRetrieveValueButton.addTarget(self, action: #selector(controlBenchmarks), for: .touchUpInside)
+            benchmarkRetrieveValueButton.setTitle("Retrieve benchmark reading", for: .normal)
+            valueMarshalWindow.rootViewController = vc
+            valueMarshalWindow.windowLevel = .init(rawValue: window!.windowLevel.rawValue + 1.0) // ???: likely not needed
+            valueMarshalWindow.isHidden = false
```

Comment:
This sets up a button to start/stop benchmarking and a text field to grab the value calculated in the app from the UI test, which can go on to perform assertions.